src/Microsoft.PowerShell.Commands.Utility/commands/utility/CSVCommands.cs

```diff
@@ -1235,9 +1241,11 @@ internal void ReadHeader()
                 TypeName = ReadTypeInformation();
             }
 
+            var values = new List<string>(valueCountGuestimate);
+            var builder = new StringBuilder(lineLengthGuestimate);
```

Comment:
We should exclude the constants here because we read a header once per file - make no sense do the optimization.

After we read the header or get it from Header parameter we know the size of the header and `values` - so we could use the size in `Import()` method below.

As for 256 const in StringBuilder I agree for the PR. Later I want add optimizations in the parser (if you already haven't such plans).

And sorry... `lineLengthGuestimate` should start with upper case `LineLengthGuestimate`.

Reply:
Why does it make more sense to do extra work when it can be avoided?